File: no_go_can_pick/easy_448_find_all_numbers_disappeared_in_an_array.js
/**
 * Easy 448. Find All Numbers Disappeared in an Array
 * https://leetcode.com/problems/find-all-numbers-disappeared-in-an-array/
 *
 * @param {number[]} nums
 * @return {number[]}
 */
var findDisappearedNumbers = function(nums) {
    let res = []
    let numsLen = nums.length
    let container = new Set()

    nums.forEach((num) => {
        container.add(num)
    });

    let arr = Array.from(container)
    let max = Math.max(...arr, numsLen)

    for (let i=1; i<=max; i++) {
        if (!container.has(i)) {
            res.push(i)
        }
    }

    return res
}

File: problems/easy_448_find_all_numbers_disappeared_in_an_array.go
package problems

import "math"

// FindDisappearedNumbers represents
// Easy 448. Find All Numbers Disappeared in an Array
// https://leetcode.com/problems/find-all-numbers-disappeared-in-an-array/
func FindDisappearedNumbers(nums []int) []int {
	var res []int
	var index int
	var key int

	for i := 0; i < len(nums); i++ {
		index = nums[i]
		key = int(math.Abs(float64(index))) - 1
		if nums[key] > 0 {
			nums[key] = -nums[key]
		}
	}

	for i := 1; i <= len(nums); i++ {
		if nums[i-1] > 0 {
			res = append(res, i)
		}
	}

	return res
}
